connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.easymock.EasyMock.replay;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class RestClientTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<TestDTO> TEST_TYPE = new TypeReference<TestDTO>() {
+    };
+    private HttpClient httpClient;
+
+    private static String toJsonString(Object obj) {
+        try {
+            return OBJECT_MAPPER.writeValueAsString(obj);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static <T> RestClient.HttpResponse<T> httpRequest(HttpClient httpClient, TypeReference<T> typeReference) {
+        return RestClient.httpRequest(
+                httpClient, null, null, null, null, typeReference, null, null);
+    }
+
+    @BeforeEach
+    public void mockSetup() {
+        httpClient = niceMock(HttpClient.class);
+    }
+
+    @Test
+    public void testSuccess() throws ExecutionException, InterruptedException, TimeoutException {
+        int statusCode = Response.Status.OK.getStatusCode();
+        String expectedResponse = toJsonString(new TestDTO("someContent"));
+        setupHttpClient(statusCode, expectedResponse);
+
+        RestClient.HttpResponse<TestDTO> httpResp = httpRequest(httpClient, TEST_TYPE);
+        assertEquals(httpResp.status(), statusCode);
+        assertEquals(toJsonString(httpResp.body()), expectedResponse);
+    }
+
+    @Test
+    public void testNoContent() throws ExecutionException, InterruptedException, TimeoutException {
+        int statusCode = Response.Status.NO_CONTENT.getStatusCode();
+        setupHttpClient(statusCode, null);
+
+        RestClient.HttpResponse<TestDTO> httpResp = httpRequest(httpClient, TEST_TYPE);
+        assertEquals(httpResp.status(), statusCode);
+        assertNull(httpResp.body());
+    }
+
+    @Test
+    public void testError() throws ExecutionException, InterruptedException, TimeoutException {
+        int statusCode = Response.Status.CONFLICT.getStatusCode();
+        ErrorMessage errorMsg = new ErrorMessage(Response.Status.GONE.getStatusCode(), "Some Error Message");
+        setupHttpClient(statusCode, toJsonString(errorMsg));
+        ConnectRestException e = assertThrows(ConnectRestException.class, () -> httpRequest(httpClient, TEST_TYPE));
+        assertEquals(e.statusCode(), statusCode);
+        assertEquals(e.errorCode(), errorMsg.errorCode());
+        assertEquals(e.getMessage(), errorMsg.message());
+    }
+
+    private void setupHttpClient(int responseCode, String responseJsonString) throws ExecutionException, InterruptedException, TimeoutException {
+        WorkerConfig workerConf = niceMock(WorkerConfig.class);
+        expect(workerConf.originals()).andReturn(Collections.emptyMap());
```

Comment:
Why are we setting up this mock? As far as I can tell it's not used where.

Reply:
true, thanks for noticing - I don't remember anymore what was the intent behind it :) 
removed